From aa7f329cc655dc4b1862fb8151ea85a832b16c5c Mon Sep 17 00:00:00 2001
From: Tianmi Chen <tianmi.chen@intel.com>
Date: Wed, 22 May 2013 10:53:21 +0800
Subject: [PORT FROM MAIN]libstagefright: release the acquired buffer when failing to read data

BZ: 106178

Release the acquired buffer when failing to read data from file to aviod
system crash.

Change-Id: Iaeebfdc8403673b5d6b48037d81d3b4c2ac93a85
Orig-Change-Id: I1a452c9f0cfbc7ca332093b73e36b3e69dd0c987
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libstagefright/AVIExtractor.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AVIExtractor.cpp b/media/libstagefright/AVIExtractor.cpp
index 0fc5754..5f17315 100644
--- a/media/libstagefright/AVIExtractor.cpp
+++ b/media/libstagefright/AVIExtractor.cpp
@@ -189,6 +189,8 @@ status_t AVIExtractor::AVISource::read(
         ssize_t n = mExtractor->mDataSource->readAt(offset, out->data(), size);
 
         if (n < (ssize_t)size) {
+            out->release();
+            out = NULL;
             return n < 0 ? (status_t)n : (status_t)ERROR_MALFORMED;
         }
 
-- 
1.7.4.1

